packages/aws-cdk/lib/commands/flag-operations.ts

```diff
@@ -233,6 +243,207 @@ async function setFlag(params: FlagOperationsParams, interactive?: boolean) {
+        await fs.remove(path.join(process.cwd(), outdir));
+        return false;
+      }
+    }
+  }
+  await fs.remove(path.join(process.cwd(), outdir));
+  return true;
+}
+
+async function isolateUnsafeFlags(
+  flags: FeatureFlag[],
+  baseContextValues: Record<string, any>,
+  toolkit: Toolkit,
+  app: string,
+  allStacks: any[],
+  queue: PQueue,
+): Promise<FeatureFlag[]> {
+  const safeFlags: FeatureFlag[] = [];
+
+  async function processBatch(batch: FeatureFlag[], contextValues: Record<string, any>): Promise<void> {
+    if (batch.length === 1) {
+      const isSafe = await testFlagSafety(batch[0], contextValues, toolkit, app, allStacks);
+      await fs.remove(path.join(process.cwd(), `test-${batch[0].name}`));
+      if (isSafe) safeFlags.push(batch[0]);
+      return;
+    }
+
+    const batchContext = { ...contextValues };
+    batch.forEach(flag => {
+      batchContext[flag.name] = toBooleanValue(flag.recommendedValue);
+    });
+
+    const isSafeBatch = await testBatch(
+      batchContext,
+      toolkit,
+      app,
+      allStacks,
+      `batch-${Date.now()}-${Math.random()}`,
+    );
+
+    if (isSafeBatch) {
+      safeFlags.push(...batch);
+      return;
+    }
+
+    const mid = Math.floor(batch.length / 2);
+    const left = batch.slice(0, mid);
+    const right = batch.slice(mid);
+
+    void queue.add(() => processBatch(left, contextValues));
+    void queue.add(() => processBatch(right, contextValues));
+  }
+
+  void queue.add(() => processBatch(flags, baseContextValues));
+
+  await queue.onIdle();
+
+  return safeFlags;
+}
+
+async function batchTestFlags(
+  flags: FeatureFlag[],
+  baseContextValues: Record<string, any>,
+  toolkit: Toolkit,
+  app: string,
+  allStacks: any[],
+  queue: PQueue,
+): Promise<FeatureFlag[]> {
+  if (flags.length === 0) {
+    return [];
+  }
+
+  const allFlagsContext = { ...baseContextValues };
+  flags.forEach(flag => {
+    allFlagsContext[flag.name] = toBooleanValue(flag.recommendedValue);
+  });
+
+  const allSafe = await testBatch(allFlagsContext, toolkit, app, allStacks, 'batch-all');
+  if (allSafe) {
+    return flags;
+  }
+
+  return isolateUnsafeFlags(flags, baseContextValues, toolkit, app, allStacks, queue);
+}
+
+async function setSafeFlags(params: FlagOperationsParams): Promise<void> {
+  const { flagData, toolkit, ioHelper, concurrency } = params;
+
+  const cdkJson = await JSON.parse(await fs.readFile(path.join(process.cwd(), 'cdk.json'), 'utf-8'));
+  const app = cdkJson.app;
+
+  const isUsingTsNode = app.includes('ts-node');
+  if (isUsingTsNode && !app.includes('-T') && !app.includes('--transpileOnly')) {
+    await ioHelper.defaults.info('You are currently running with ts-node. Adding --transpileOnly may make this operation faster.');
+  }
+
+  const unconfiguredFlags = flagData.filter(flag =>
+    flag.userValue === undefined &&
+    isBooleanFlag(flag) &&
+    (flag.unconfiguredBehavesLike?.v2 !== flag.recommendedValue),
```

Comment:
```suggestion
    ((flag.unconfiguredBehavesLike?.v2 ?? false) !== flag.recommendedValue),
```

All the more reason to get this from a single place.